### ISSUE TITLE
dbapi_hook get_records doesn't accept list.

### ISSUE THREAD

**Issue by JCoder01** [apache/airflow#9957]
**Apache Airflow version**:1.10.10

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

passing a list to `dbapi_hook.get_records` fails
`TypeError: argument 1 must be a string or unicode object: got list instead`

**What you expected to happen**:

return list of query results.

**How to reproduce it**:
```
from airflow.hooks.postgres_hook import PostgresHook
PostgresHook().get_records(['select * from dag_run limit 1', 'select * from dag_run limit 1'])
```

**Anything else we need to know**:

**Comment by jhtimmins** [comment]
I was able to replicate this bug. Sqlalchemy expects to receive a string, not a list. It would be trivial to iterate over the input statements and run them in sequence.

However, it's not clear to me what the original plan for this method was, since it accepts a list or dictionary of parameters. SqlAlchemy has a pattern in which a parameterized query string is passed in, and then a list of dicts/lists containing parameters is as passed well. The query is run against all of the parameters individually. But if `get_records` accept both a list of queries _and_ a list of parameters, it's not clear if or how the parameters should be applied to individual query statements.

@kaxil what do you think?

**Comment by kaxil** [comment]
Based on https://www.psycopg.org/docs/cursor.html#cursor.execute and https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlcursor-execute.html it looks like the cursor.execute method is up[posed to just take a string (query to execute).

**Comment by JCoder01** [comment]
@jhtimmins is your question whether we should iterate over the sql statements and the parameters, or just iterate over the sql statements and pass in the same parameters to each sql item? My belief is that it should be the latter, and how it has been implemented in the PR.

**Comment by pkonarzewski** [comment]
I'd like to try to fix it

**Comment by mik-laj** [comment]
@pkonarzewski I assigned you to this ticket 🐈

**Comment by pkonarzewski** [comment]
So I've wrote some code to fix this issue, but now i have some doubts.
It's a simple fix, but there is few corner cases, that make it a little bit messy for me.

My problems are:
1. get_records() with one sql returns List[Tuple], with list of sqls it will return List[List[Tuple]], is this inconsistency ok?
2. get_first() returns Tuple, should multi sql statements return List[Tuple]? Same problem as in point 1.
3. what to do when one of the queries from list returns empty result?
4. there is also method get_pandas_df(), shouldn't it also allow multiple sql statements?

What do you think?

**Comment by kazanzhy** [comment]
@eladkal 
When we moved to the separate `common.sql` provider it was untyped.
After this commit, these types were added https://github.com/apache/airflow/pull/25299/files#diff-6e1b2f961cb951d05d66d2d814ef5f6d8f8bf8f43c40fb5d40e27a031fed8dd7

Oh, I just saw that it's a 2-yeas old issue.
There's a simple `execute` method used here, therefore I think we have the wrong type here.
At least for Postgres https://www.psycopg.org/docs/cursor.html#execute

We could try to use `DbApiHook.run` for the `DbApiHook.get_records` and `DbApiHook.get_first`. 
I'll try to implement it
